freqtrade/freqtradebot.py

```diff
@@ -443,33 +442,28 @@ def process_maybe_execute_buy(self) -> None:
         except DependencyException as exception:
             logger.warning('Unable to create trade: %s', exception)
 
-    def process_maybe_execute_sell(self, trade: Trade) -> bool:
+    def process_maybe_execute_sells(self, trades: List[Any]) -> None:
         """
-        Tries to execute a sell trade
-        :return: True if executed
+        Tries to execute sell trades in a safe way
         """
-        try:
-            self.update_trade_state(trade)
-
-            if self.strategy.order_types.get('stoploss_on_exchange') and trade.is_open:
-                result = self.handle_stoploss_on_exchange(trade)
-                if result:
-                    self.wallets.update()
-                    return result
-
-            if trade.is_open and trade.open_order_id is None:
-                # Check if we can sell our current pair
-                result = self.handle_trade(trade)
-
-                # Updating wallets if any trade occured
-                if result:
-                    self.wallets.update()
-
-                return result
-
-        except DependencyException as exception:
-            logger.warning('Unable to sell trade: %s', exception)
-        return False
+        for trade in trades:
+            try:
+                self.update_trade_state(trade)
+
+                if trade.is_open:
+                    result = False
+                    if self.strategy.order_types.get('stoploss_on_exchange'):
+                        result = self.handle_stoploss_on_exchange(trade)
+                    elif trade.open_order_id is None:
```

Comment:
This is wrong (as you discovered already ...)
even if `stoploss_on_exchange` is enabled, we still need to `handle_trade()` - otherwise `stoploss_on_exchange` remains the only sell-reason that's technically possible if it's enabled.
You can use `continue` if stoploss_on_exchanged fired instead of elif i guess ... (in combination with the other comment a few lines above ...)

What surprises me more than this small issue is that no test complains about this.

Reply:
Got it, fixed the logic, see below...